source/game_sa/Tasks/TaskTypes/TaskSimpleKillPedWithCar.cpp

```diff
@@ -0,0 +1,172 @@
+
+    RH_ScopedInstall(Constructor, 0x653C10);
+    RH_ScopedInstall(Destructor, 0x653C90);
+
+    RH_ScopedInstall(DamageCarBonnet, 0x653CF0);
+    RH_ScopedVMTInstall(Clone, 0x655A20);
+    RH_ScopedVMTInstall(GetTaskType, 0x653C70);
+    RH_ScopedVMTInstall(MakeAbortable, 0x653C80);
+    RH_ScopedVMTInstall(ProcessPed, 0x6564A0);
+}
+
+// 0x653C10
+CTaskSimpleKillPedWithCar::CTaskSimpleKillPedWithCar(CVehicle* car, float impulseMag) :
+    m_Car{car},
+    m_ImpulseMag{impulseMag}
+{
+}
+
+// 0x655A20
+CTaskSimpleKillPedWithCar::CTaskSimpleKillPedWithCar(const CTaskSimpleKillPedWithCar& o) :
+    CTaskSimpleKillPedWithCar{o.m_Car, o.m_ImpulseMag}
+{
+}
+
+// 0x653CF0
+void CTaskSimpleKillPedWithCar::DamageCarBonnet(CPed const*) {
+    if (!m_Car->IsSubAutomobile()) {
+        return;
+    }
+
+    const auto flyingObj = m_Car->AsAutomobile()->RemoveBonnetInPedCollision();
+    if (!flyingObj) {
+        return;
+    }
+
+    const auto dir = m_Car->GetRight() * 0.1f + m_Car->GetUp() * 0.5f;
+    flyingObj->m_vecMoveSpeed += CGeneral::DoCoinFlip() ? dir : -dir;
+
+    flyingObj->ApplyTurnForce(m_Car->GetUp() * 10.f, m_Car->GetForward());
+}
+
+// 0x6564A0
+bool CTaskSimpleKillPedWithCar::ProcessPed(CPed* ped) {
+    const auto  carPos        = m_Car->GetPosition();
+    const auto  carCM         = m_Car->GetColModel();
+    const auto& carBB         = carCM->GetBoundingBox();
+    const auto& carVel        = m_Car->GetMoveSpeed();
+    const auto& carFwd        = m_Car->GetForward();
+    const auto  carVelFwd     = carFwd.Dot(carVel); // Velocity forwards
+    const auto  carIsGoingFwd = carVelFwd >= 0.f;
+
+    const auto pedPos     = ped->GetPosition();
+    const auto pedToCar   = pedPos - carPos;
+    const auto pedPosOS   = Multiply3x3(pedToCar, m_Car->GetMatrix()); // Ped's position in the Car's object space
+    auto       pedHitSide = (eDirection)CPedGeometryAnalyser::ComputePedHitSide(*ped, *m_Car);
+
+    eWeaponType killWT = WEAPON_RAMMEDBYCAR;
+    enum {
+        FROM_LEFT = 1,
+        FROM_RIGHT = 3,
+        FROM_BACK = 4,
+        FROM_FRONT = 5
+    } carDir{};
+    if (std::abs(pedPosOS.x) > carBB.m_vecMax.x * 0.99f) { // 0x6565D5 - Ped is on left/right (This assumes abs(bb.max.x) == abs(bb.min.x))
+        carDir = pedPosOS.x < 0.f
+            ? FROM_LEFT
+            : FROM_RIGHT;
+        if (std::abs(pedPosOS.y) < std::abs((carIsGoingFwd ? carBB.m_vecMax.y : carBB.m_vecMin.y) * 0.85f)) { // Did we run over the ped?
+            killWT = WEAPON_RUNOVERBYCAR;
+
+            ped->m_vecMoveSpeed = CVector{CVector2D{carVel} * 0.9f};
+            if (CLocalisation::KnockDownPeds()) {
+                m_Car->m_vehicleAudio.AddAudioEvent(AE_PED_CRUNCH, 0.f);
+            }
+
+            switch (pedHitSide) { // 0x6566A7
+            case eDirection::LEFT:
+            case eDirection::RIGHT:
+                pedHitSide = eDirection::FORWARD;
+            }
+        }
+    } else if (pedPosOS.z > 0.1f) { // 0x6566CB
```

Comment:
to improve comments, though tbh it was a good recap for linear algebra =D
```suggestion
    } else if (pedPosOS.z > 0.1f) { // 0x6566CB -- ped is on the bonnet
```

Reply:
It's not necessarily on the bonnet tho, no?
It's just under/over the car i think

Reply:
> It's not necessarily on the bonnet tho, no? It's just under/over the car i think

there is a `DamageCarBonnet` call at the end tho.